src/datasources/system/SystemDataSource.ts

```diff
@@ -32,7 +33,8 @@ export class SystemDataSource extends DataSourceApi<SystemQuery> {
           ],
         });
       } else {
-        throw Error("Not implemented");
+        let metadataResponse = await getBackendSrv().post<{ data: SystemMetadata[] }>(this.baseUrl + '/query-systems', { projection: "new(id, alias, connected.data.state, grains.data.minion_blackout as locked, grains.data.boottime as systemStartTime, grains.data.productname as model, grains.data.manufacturer as vendor, grains.data.osfullname as osFullName, grains.data.ip4_interfaces as ip4Interfaces, grains.data.ip6_interfaces as ip6Interfaces, workspace)" });
```

Comment:
Is there a better way we can construct this JSON string to help with readability?

Reply:
I don't think so? I tried putting each projection on a different line, but it gave me a bunch of errors/made the format weird. @mure do you have any suggestions?

Reply:
I think I'd extract this to a new file `constants.ts` and turn it into an array.
```ts
// constants.ts
export const defaultProjection = [
  'id',
  'alias',
  'connected.data.state'
]
```

and then add a method to the data source that transforms the array into the `"new(...)"` string.

Reply:
Also, we needed to add some logic to pull out the first ip address. Is that work tracked anywhere? We should add another user story or add it to the acceptance criteria of an existing one.